tests/test_raster.py

```diff
@@ -596,6 +597,81 @@ def test_data_setter(self, dtype: str, nodata_init: str | None) -> None:
+
+        raster_area = gu.Raster(self.aster_dem_path)
+        assert raster_area.area_or_point == "Area"
+
+        # 2/ Setter
+        # For None, it will remove the key from the tags dictionary
+        raster_point.area_or_point = None
+        assert raster_point.area_or_point is None
+        assert "AREA_OR_POINT" not in raster_point.tags
+
+        # For a good value, it will update the tags
+        raster_point.area_or_point = "Point"
+        assert raster_point.area_or_point == "Point"
+        assert "AREA_OR_POINT" in raster_point.tags and raster_point.tags["AREA_OR_POINT"] == "Point"
+
+        # 2.5/ Setter shift and using set_area_or_point() with options
+        old_transform = raster_point.transform
+        # Setting "Area" from "Point" will shift the transform by default
+        raster_point.area_or_point = "Area"
+        assert raster_point.area_or_point == "Area"
+        assert raster_point.transform != old_transform
+
+        # Setting "Pixel" should give back the original transform
+        raster_point.area_or_point = "Point"
+        assert raster_point.area_or_point == "Point"
+        assert raster_point.transform == old_transform
+
+        # Setting None or "Pixel" should trigger no shift
+        raster_point.area_or_point = "Point"
+        assert raster_point.area_or_point == "Point"
+        assert raster_point.transform == old_transform
+        raster_point.area_or_point = None
+        assert raster_point.area_or_point is None
+        assert raster_point.transform == old_transform
+
+        # Setting "Area" passing a shift argument of False also shouldn't
+        raster_point.set_area_or_point("Area", shift_area_or_point=False)
+        assert raster_point.area_or_point == "Area"
+        assert raster_point.transform == old_transform
+
+        # Setting "Area" with a globally False argument of shift_area_or_point also shouldn't
+        geoutils.config["shift_area_or_point"] = False
+        raster_point.set_area_or_point("Point")
+        assert raster_point.area_or_point == "Point"
+        assert raster_point.transform == old_transform
+        # We reset the config argument
+        geoutils.config["shift_area_or_point"] = True
+
+        # 3/ With function creating a single Raster
+
+        # From array
+        raster_point_fromarray = gu.Raster.from_array(
+            data=raster_point.data,
+            transform=raster_point.transform,
+            area_or_point=raster_point.area_or_point,
+            crs=raster_point.crs,
+        )
+        assert raster_point.area_or_point == raster_point_fromarray.area_or_point
+
+        # Copy
+        raster_point_copy = raster_point.copy()
+        assert raster_point.area_or_point == raster_point_copy.area_or_point
+
+        # 4/ With function casting from several Rasters
+
+        # with pytest.warns(UserWarning, match='One raster has a pixel interpretation "Area" and the other "Point".*'):
+        #     r
```

Comment:
Some missing code here?

Reply:
Ah yes, hadn't finished this! I just added the tests for these raised warnings in the `TestArithmetic` and `TestArrayInterface` functions `test_raise_errors`, where they can use the parametrized loops and be more thorough :)